### PR TITLE
Refactor error handled forms

### PR DESCRIPTION
**Pull request by alexmatson-da**

### CODE REVIEW

File: ui/src/components/common/Holdings.tsx [hunk 1]
```diff
@@ -73,30 +72,27 @@ type FormProps = {
...
-                await ledger.exerciseByKey(Broker.Broker_AllocateToProvider, key, args)
-            }
-            clearForm()
-        } catch (err) {
-            setError(parseError(err));
+                break;
+            case MarketRole.BrokerRole:
+                await ledger.exerciseByKey(Broker.Broker_AllocateToProvider, key, args);
+                break;
+            default:
```
Comment (dimitri-da) [inline ui/src/components/common/Holdings.tsx:91]: nice

File: ui/src/components/Broker/BrokerOrders.tsx [hunk 2]
```diff
@@ -136,58 +118,42 @@ type BrokerOrderCardProps = {
...
+        text: `${deposit.contractData.asset.quantity} ${deposit.contractData.asset.id.label} | ${deposit.contractData.account.id.label}`
+    }))
 
-    function clearForm() {
-        setDepositCid('')
+    const handleAcceptBrokerOrderFill = async () => {
+        const key = wrapDamlTuple([props.cdata.broker, props.cdata.brokerOrderId])
+        const args = { depositCid }
+        await ledger.exerciseByKey(BrokerOrder.BrokerOrder_Fill, key, args);
+        setDepositCid('');
```
Comment (dimitri-da) [inline ui/src/components/Broker/BrokerOrders.tsx:144]: should we use `.then(setDepositCid(''))` here and in all other places where we clear the forms? (ie clear the form only on successful submission). The reasoning is that if the user inputed something that was wrong then hit submit, they would still have the option to correct it inline.

File: ui/src/components/Custodian/CreateDeposit.tsx [hunk 3]
```diff
@@ -26,37 +19,26 @@ const CreateDeposit: React.FC = () => {
...
 
-    function clearForm() {
+        const tokenId = token.contractData.id;
+        const args = { beneficiary, tokenId, depositQuantity };
+        const key = wrapDamlTuple([operator, custodian]);
+        await ledger.exerciseByKey(Custodian.Custodian_CreateDeposit, key, args);
+
         setBeneficiary('')
         setToken(undefined)
         setDepositQuantity('')
```
Comment (dimitri-da) [inline ui/src/components/Custodian/CreateDeposit.tsx:37]: same as above here

File: ui/src/components/Exchange/CreateMarket.tsx [hunk 4]
```diff
@@ -22,44 +21,31 @@ type Props = {
...
-            await ledger.exerciseByKey(Exchange.Exchange_AddPair, key, args);
-            clearForm();
-        } catch (err) {
-            setError(parseError(err));
-        }
-        setLoading(false);
+        await ledger.exerciseByKey(Exchange.Exchange_AddPair, key, args);
+
+        setBaseToken(undefined);
+        setQuoteToken(undefined);
```
Comment (dimitri-da) [inline ui/src/components/Exchange/CreateMarket.tsx:48]: here as well. (you get the idea)

**Review by dimitri-da — APPROVED** [review]
looks good to me. Just a suggestion on field behavior when we have an error

File: ui/src/components/Broker/BrokerOrders.tsx (hunk 2 above)
Comment (alexmatson-da) [reply ui/src/components/Broker/BrokerOrders.tsx:144]: Ah, this already happens as you describe. Everything in this function gets wrapped in a `try { } except { }`, so the execution of this expands out to 

```javascript
setLoading(true);
try {
        const key = wrapDamlTuple([props.cdata.broker, props.cdata.brokerOrderId])
        const args = { depositCid }
        await ledger.exerciseByKey(BrokerOrder.BrokerOrder_Fill, key, args);
        setDepositCid('');
} catch (err) {
        setError(parseError(err));
}
setLoading(false);
```

inside the FormErrorHandled.

So if an error is caught from the ledger API call, then it won't clear

**Review by alexgraham-da — APPROVED** [review]

**Pull request merged**